src/main/scala/cromwell/util/docker/DockerIdentifierParser.scala

```diff
@@ -0,0 +1,68 @@
+package cromwell.util.docker
+
+import scala.annotation.tailrec
+
+object DockerIdentifierParser {
+  def parse(identifier: String): DockerIdentifier = parse(identifier, parsers)
+
+  private val parsers = Seq(GcrDockerParser, DockerHubParser, DockerHubLibraryParser)
+
+  @tailrec
+  private def parse(identifier: String, parsers: Seq[DockerIdentifierParser]): DockerIdentifier = {
+    parsers.headOption match {
+      case Some(parser) =>
+        val tryMatch = parser.matchIdentifier
+        if (tryMatch.isDefinedAt(identifier)) tryMatch.apply(identifier) else parse(identifier, parsers.tail)
+      case None => throw new Error("Docker Hub Library Parser should have matched.")
```

Comment:
The `DockerHubLibraryParser.hubLibraryLatest` is a catch all regexp, meant to catch things like `docker pull ubuntu`. Actually, I should probably just explicitly handle that here instead of throwing the `Error`.